bot/exts/moderation/infraction/infractions.py

```diff
@@ -664,6 +645,31 @@ async def cog_command_error(self, ctx: Context, error: Exception) -> None:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """
+
+        Apply active timeout infractions for returning members.
+
+        This is needed for users who might have had their infraction edited in our database but not in Discord itself.
+        """
+        active_timeouts = await self.bot.api_client.get(
+            endpoint="bot/infractions",
+            params={"active": "true", "type": "timeout", "user__id": member.id}
+        )
+
+        if active_timeouts:
+            timeout_infraction = active_timeouts[0]
+            expiry = arrow.get(timeout_infraction["expires_at"], tzinfo=UTC).datetime
+            if member.is_timed_out():
+                expiry = max(expiry, member.timed_out_until)
```

Comment:
Shouldn't we always use the expiry from the database?
With your current approach, if a moderator reduces the timeout duration, the infraction log will display the new duration. However, when the user rejoins the server, the old (original) longer timeout duration will be applied.

Reply:
I'm not on a streak this period of the year...

You're right, I'll fix that.